internal/ignition/ignition.go

```diff
@@ -1403,6 +1406,17 @@ func (ib *ignitionBuilder) FormatDiscoveryIgnitionFile(ctx context.Context, infr
 		ib.log.WithError(err).Errorln("Unable to build user SSH public key JSON")
 		return "", err
 	}
+	// handle ZTP converged flow
+	UseConvergedFlow := false
+	if infraEnv.EnableIronicAgent {
```

Comment:
there is no validation in infraenv controller for it? not sure it is very obvious place to verify infra env  validity

Reply:
I'm assuming that if the user requested converged but converged isn't available, this error would be propagated to the InfraEnv Status in some way?